src/System.CommandLine/ConvertArgument.cs

```diff
@@ -3,5 +3,7 @@
 
 namespace System.CommandLine
 {
-    public delegate ArgumentResult ConvertArgument(SymbolResult symbolResult);
+    public delegate bool TryConvertArgument(SymbolResult symbolResult, out object value);
+
+    public delegate bool TryConvertArgument<T>(SymbolResult symbolResult, out T value);
```

Comment:
IMO I liked the delegate signature better when it was expected to return a result object, and that object handled the success/failure rather than going with the TryParse pattern

Reply:
The reason I made this change was that it felt a bit more discoverable and arguably more idiomatic for C#. This also helps to untangle a chicken/egg problem between parsing and conversion/validation. That's very much a work in progress though.